nav2_map_server/src/map_io.cpp

```diff
@@ -261,12 +261,25 @@ void loadMapFromFile(
       normalized = (1.0f - normalized.array()).matrix();
     }
 
+    double occupied_thresh = load_parameters.occupied_thresh;
+    double free_thresh = load_parameters.free_thresh;
+
+    // For Trinary mode, the thresholds depend on the constants used when saving the map,
+    // not on the thresholds from the yaml file
+    // occupied is always 1.0, free is always 0.00392157 ( = 1.0 - 254 / 255.0),
+    // and unknown is always 0.196078431 ( = 1.0 - 205 / 255.0)
+    // For Scale mode, the thresholds are set by the user in the yaml file
+    if (load_parameters.mode == MapMode::Trinary) {
```

Comment:
I dont' see this as consistent with the previous behavior https://github.com/ros-navigation/navigation2/blob/jazzy/nav2_map_server/src/map_io.cpp#L254-L260

Reply:
The behavior in Jazzy has the same issue. I was planning to fix it for Jazzy, even before the new Eigen PR

Reply:
Can you point to where this is in ROS 1 then?

Reply:
Loading: https://github.com/ros-planning/navigation/blob/noetic-devel/map_server/src/image_loader.cpp
Saving pixels: https://github.com/ros-planning/navigation/blob/noetic-devel/map_server/src/map_saver.cpp#L77-L82
Saving yaml file: https://github.com/ros-planning/navigation/blob/noetic-devel/map_server/src/map_saver.cpp#L96-L116

In ROS1 they use the 2nd option I proposed, i.e. always saving `occupied_thresh: 0.65, free_thresh: 0.196` regardless of the thresholds used to threshold the map (given by the user with `--occ` and `--free`

Reply:
In ROS 2 we use the thresholds the user set though https://github.com/ros-navigation/navigation2/blob/main/nav2_map_server/src/map_saver/map_saver.cpp

Reply:
Yes, but those are used to threshold the probability map. They are also used in ROS 1 [here](https://github.com/ros-planning/navigation/blob/noetic-devel/map_server/src/map_saver.cpp#L77-L82)
But ROS1 doesn't save these to file, instead always saving 0.196 & 0.65.